internal/moreos/moreos_func-e_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package moreos
+
+import (
+	"bytes"
+	"embed"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/tetratelabs/func-e/internal/test/fakebinary"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	// fakeFuncESrc is a test source file used to simulate how func-e manages its child process
+	//go:embed testdata/fake_func-e.go
+	fakeFuncESrc []byte // Embedding the fakeFuncESrc is easier than file I/O and ensures it doesn't skew coverage
+
+	// Include the source imported by fakeFuncESrc directly and indirectly
+	//go:embed moreos.go
+	//go:embed proc_*.go
+	moreosSrcDir embed.FS
+)
+
+// Test_X tests multiple features include moreos.EnsureProcessDone and moreos.ProcessGroupAttr to ensure X occurs when Y
+func Test_X(t *testing.T) {
```

Comment:
Test_X was a placeholder to weave in TestProcessGroupAttr_Kill :D

Reply:
Oops

Reply:
re: `Test_X` do we want a separated test for when func-e successfully running envoy vs immediate exit?

Reply:
we can add this stuff back later, if we need it. For the sake of this change, we can focus on the most primary case of interest, which is signal propagation!